### PR TITLE
chore: refactor chromium's `urlToWSEndpoint`

### PR DESCRIPTION
**Pull request by aslushnikov**

### CODE REVIEW

**Review by mxschmitt — APPROVED** [review]

File: packages/playwright-core/src/server/chromium/chromium.ts [hunk 1]
```diff
@@ -364,20 +361,16 @@ async function urlToWSEndpoint(progress: Progress, endpointURL: string) {
...
-      let data = '';
-      resp.on('data', chunk => data += chunk);
-      resp.on('end', () => resolve(data));
-    }).on('error', reject);
-  });
+  const onError = async (params: HTTPRequestParams, response: http.IncomingMessage) => {
+    return new Error(`Unexpected status ${response.statusCode} when connecting to ${httpURL}.\n` +
+        `This does not look like a DevTools server, try connecting via ws://.`);
+  };
+  const json = await fetchData({
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/chromium/chromium.ts:368]: drive-by fix. You want to make sure that it contains the playwright user-agent.

**Comment by aslushnikov** [comment]
We decided not to touch this at all.
